otwcode/otwarchive
# AO3-5910 Update schema and structure files

## Issue

https://otwarchive.atlassian.net/browse/AO3-5910

## Purpose

Generate new versions of schema dumps from production.

For Ruby dump, run on an app server:
```sh
bundle exec rake db:schema:dump
```
then manually change the version in `ActiveRecord::Schema.define` to the latest migration timestamp.

For SQL dump, run on a database server:
```sh
mysqldump --no-data --routines --skip-comments --result-file=structure.sql otwarchive_production
```
```
Warning: A partial dump from a server that has GTIDs will by default include the GTIDs of all transactions, even those that changed suppressed parts of the database. If you don't want to restore GTIDs, pass --set-gtid-purged=OFF. To make a complete dump, pass --all-databases --triggers --routines --events.
```
then manually:
- remove [AUTO_INCREMENT values](https://stackoverflow.com/questions/2210719/out-of-sync-auto-increment-values-in-development-structure-sql-from-rails-mysql);
- append a dump of the schema_migrations table, generated from `bundle exec rails r "puts ActiveRecord::Base.connection.dump_schema_information"`.

The resulting output will be similar to that of `bundle exec rake db:structure:dump` (which we can't use because we've started provisioning app and database servers differently, and app servers [no longer have mysqldump available](https://otwarchive.atlassian.net/browse/AO3-5928)).

## Testing Instructions

CI build should pass.

After this is merged:
- Check the existing pull requests for migrations and [bump their timestamps to later](https://stackoverflow.com/a/4047884).
- Update https://github.com/otwcode/otwarchive/wiki/Migrations#updating-schema-dumps.
